Panic in kubectl jsonpath templating

```console
$ kubectl config view -o 'jsonpath={.contexts[?(@.name == \"\")].context.user}'
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x10 pc=0xbbf926]
goroutine 1 [running]:
k8s.io/kubernetes/pkg/util/jsonpath.parseAction(0x13284f0, 0x5, 0xc2080b0edf, 0x5, 0xc20800b450, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/jsonpath/parser.go:66 +0x266
k8s.io/kubernetes/pkg/util/jsonpath.(*Parser).parseFilter(0xc20800b3b0, 0xc208180ea0, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/jsonpath/parser.go:349 +0x712
k8s.io/kubernetes/pkg/util/jsonpath.*Parser.(k8s.io/kubernetes/pkg/util/jsonpath.parseFilter)·fm(0xc208180ea0, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/jsonpath/parser.go:140 +0x43
k8s.io/kubernetes/pkg/util/jsonpath.(*Parser).parseInsideAction(0xc20800b3b0, 0xc208180ea0, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/jsonpath/parser.go:145 +0x3a0
k8s.io/kubernetes/pkg/util/jsonpath.(*Parser).parseField(0xc20800b3b0, 0xc208180ea0, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/jsonpath/parser.go:391 +0x359
k8s.io/kubernetes/pkg/util/jsonpath.(*Parser).parseInsideAction(0xc20800b3b0, 0xc208180ea0, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/jsonpath/parser.go:161 +0x55b
k8s.io/kubernetes/pkg/util/jsonpath.(*Parser).parseLeftDelim(0xc20800b3b0, 0xc208180ea0, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/jsonpath/parser.go:134 +0x277
k8s.io/kubernetes/pkg/util/jsonpath.(*Parser).parseText(0xc20800b3b0, 0xc208180e80, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/jsonpath/parser.go:114 +0x3bc
k8s.io/kubernetes/pkg/util/jsonpath.(*Parser).Parse(0xc20800b3b0, 0xc2080b0ec9, 0x2b, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/jsonpath/parser.go:74 +0xc8
k8s.io/kubernetes/pkg/util/jsonpath.Parse(0x13258f0, 0x3, 0xc2080b0ec9, 0x2b, 0x1274660, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/jsonpath/parser.go:50 +0xbc
k8s.io/kubernetes/pkg/util/jsonpath.(*JSONPath).Parse(0xc208178600, 0xc2080b0ec9, 0x2b, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/util/jsonpath/jsonpath.go:50 +0x5e
k8s.io/kubernetes/pkg/kubectl.NewJSONPathPrinter(0xc2080b0ec9, 0x2b, 0x1c0e298, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource_printer.go:1672 +0xdf
k8s.io/kubernetes/pkg/kubectl.GetPrinter(0x135d450, 0x8, 0xc2080b0ec9, 0x2b, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource_printer.go:96 +0x11b6
k8s.io/kubernetes/pkg/kubectl/cmd/util.PrinterForCommand(0xc208175c20, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/util/printing.go:106 +0x305
k8s.io/kubernetes/pkg/kubectl/cmd/config.func·008(0xc208175c20, 0xc208180e60, 0x0, 0x2)
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/config/view.go:64 +0x51
github.com/spf13/cobra.(*Command).execute(0xc208175c20, 0xc208180ca0, 0x2, 0x2, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/spf13/cobra/command.go:547 +0x82f
github.com/spf13/cobra.(*Command).Execute(0xc2080def00, 0x0, 0x0)
        /go/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/spf13/cobra/command.go:630 +0x414
main.main()
        /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:30 +0x104
goroutine 5 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x1c02040)
        /go/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x78
created by github.com/golang/glog.init·1
        /go/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x2a7
goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/src/go/src/runtime/asm_amd64.s:2232 +0x1
goroutine 13 [syscall]:
os/signal.loop()
        /usr/src/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
        /usr/src/go/src/os/signal/signal_unix.go:27 +0x35
```

@kubernetes/goog-ux

--- a/pkg/kubectl/cmd/config/view.go
+++ b/pkg/kubectl/cmd/config/view.go
@@ -18,6 +18,7 @@ package config
 
 import (
 	"errors"
+	"fmt"
 	"io"
 
 	"github.com/golang/glog"
@@ -52,6 +53,8 @@ $ kubectl config view -o template --template='{{range .users}}{{ if eq .name "e2
 
 func NewCmdConfigView(out io.Writer, ConfigAccess ConfigAccess) *cobra.Command {
 	options := &ViewOptions{ConfigAccess: ConfigAccess}
+	// Default to yaml
+	defaultOutputFormat := "yaml"
 
 	cmd := &cobra.Command{
 		Use:     "view",
@@ -60,6 +63,11 @@ func NewCmdConfigView(out io.Writer, ConfigAccess ConfigAccess) *cobra.Command {
 		Example: view_example,
 		Run: func(cmd *cobra.Command, args []string) {
 			options.Complete()
+			outputFormat := cmdutil.GetFlagString(cmd, "output")
+			if outputFormat == "wide" {
+				fmt.Printf("--output wide is not available in kubectl config view; reset to default output format (%s)\n\n", defaultOutputFormat)
+				cmd.Flags().Set("output", defaultOutputFormat)
+			}
 
 			printer, _, err := cmdutil.PrinterForCommand(cmd)
 			if err != nil {
@@ -76,8 +84,7 @@ func NewCmdConfigView(out io.Writer, ConfigAccess ConfigAccess) *cobra.Command {
 	}
 
 	cmdutil.AddPrinterFlags(cmd)
-	// Default to yaml
-	cmd.Flags().Set("output", "yaml")
+	cmd.Flags().Set("output", defaultOutputFormat)
 
 	options.Merge.Default(true)
 	cmd.Flags().Var(&options.Merge, "merge", "merge together the full hierarchy of kubeconfig files")

--- a/pkg/util/jsonpath/parser.go
+++ b/pkg/util/jsonpath/parser.go
@@ -63,8 +63,12 @@ func NewParser(name string) *Parser {
 // parseAction parsed the expression inside delimiter
 func parseAction(name, text string) (*Parser, error) {
 	p, err := Parse(name, fmt.Sprintf("%s%s%s", leftDelim, text, rightDelim))
+	// when error happens, p will be nil, so we need to return here
+	if err != nil {
+		return p, err
+	}
 	p.Root = p.Root.Nodes[0].(*ListNode)
-	return p, err
+	return p, nil
 }
 
 func (p *Parser) Parse(text string) error {

--- a/pkg/util/jsonpath/parser_test.go
+++ b/pkg/util/jsonpath/parser_test.go
@@ -21,44 +21,46 @@ import (
 )
 
 type parserTest struct {
-	name  string
-	text  string
-	nodes []Node
+	name        string
+	text        string
+	nodes       []Node
+	shouldError bool
 }
 
 var parserTests = []parserTest{
-	{"plain", `hello jsonpath`, []Node{newText("hello jsonpath")}},
+	{"plain", `hello jsonpath`, []Node{newText("hello jsonpath")}, false},
 	{"variable", `hello {.jsonpath}`,
-		[]Node{newText("hello "), newList(), newField("jsonpath")}},
+		[]Node{newText("hello "), newList(), newField("jsonpath")}, false},
 	{"arrayfiled", `hello {['jsonpath']}`,
-		[]Node{newText("hello "), newList(), newField("jsonpath")}},
-	{"quote", `{"{"}`, []Node{newList(), newText("{")}},
+		[]Node{newText("hello "), newList(), newField("jsonpath")}, false},
+	{"quote", `{"{"}`, []Node{newList(), newText("{")}, false},
 	{"array", `{[1:3]}`, []Node{newList(),
-		newArray([3]ParamsEntry{{1, true}, {3, true}, {0, false}})}},
+		newArray([3]ParamsEntry{{1, true}, {3, true}, {0, false}})}, false},
 	{"allarray", `{.book[*].author}`,
 		[]Node{newList(), newField("book"),
-			newArray([3]ParamsEntry{{0, false}, {0, false}, {0, false}}), newField("author")}},
+			newArray([3]ParamsEntry{{0, false}, {0, false}, {0, false}}), newField("author")}, false},
 	{"wildcard", `{.bicycle.*}`,
-		[]Node{newList(), newField("bicycle"), newWildcard()}},
+		[]Node{newList(), newField("bicycle"), newWildcard()}, false},
 	{"filter", `{[?(@.price<3)]}`,
 		[]Node{newList(), newFilter(newList(), newList(), "<"),
-			newList(), newField("price"), newList(), newInt(3)}},
-	{"recursive", `{..}`, []Node{newList(), newRecursive()}},
+			newList(), newField("price"), newList(), newInt(3)}, false},
+	{"recursive", `{..}`, []Node{newList(), newRecursive()}, false},
 	{"recurField", `{..price}`,
-		[]Node{newList(), newRecursive(), newField("price")}},
+		[]Node{newList(), newRecursive(), newField("price")}, false},
 	{"arraydict", `{['book.price']}`, []Node{newList(),
 		newField("book"), newField("price"),
-	}},
+	}, false},
 	{"union", `{['bicycle.price', 3, 'book.price']}`, []Node{newList(), newUnion([]*ListNode{}),
 		newList(), newField("bicycle"), newField("price"),
 		newList(), newArray([3]ParamsEntry{{3, true}, {4, true}, {0, false}}),
 		newList(), newField("book"), newField("price"),
-	}},
+	}, false},
 	{"range", `{range .items}{.name},{end}`, []Node{
 		newList(), newIdentifier("range"), newField("items"),
 		newList(), newField("name"), newText(","),
 		newList(), newIdentifier("end"),
-	}},
+	}, false},
+	{"malformat input", `{\\\}`, []Node{}, true},
 }
 
 func collectNode(nodes []Node, cur Node) []Node {
@@ -82,6 +84,12 @@ func collectNode(nodes []Node, cur Node) []Node {
 func TestParser(t *testing.T) {
 	for _, test := range parserTests {
 		parser, err := Parse(test.name, test.text)
+		if test.shouldError {
+			if err == nil {
+				t.Errorf("unexpected non-error when parsing %s", test.name)
+			}
+			continue
+		}
 		if err != nil {
 			t.Errorf("parse %s error %v", test.name, err)
 		}

